inat_finder.py

```
@@ -0,0 +1,589 @@
+    This function orchestrates the search for valid iNaturalist observations by:
+    - Parsing command-line arguments and extracting an observation number from a URL or plain input.
+    - Verifying that the specified genus or username exists in iNaturalist.
+    - Validating the observation number and warning the user if it appears too short (suggesting a possible Mushroom Observer observation).
+    - Optionally confirming whether the original observation number already matches the target genus or username.
+    - Generating possible variations by altering digits and appending additional ones for shorter numbers.
+    - Checking these variations in batches via API calls and displaying progress.
+    - Presenting a summary of potential matches and the overall search duration.
+
+    Note: This function interacts with the user via input prompts and exits if critical validation fails.
+    """
+    BATCH_SIZE = 200  # Define batch size as a constant
```

Pylint (3.3.7)

[convention] Variable name "BATCH_SIZE" doesn't conform to snake_case naming style

(C0103)